icu-date.lua

```diff
@@ -61,49 +61,73 @@ local function uchar_to_string(uchar)
   return ffi.string(str)
 end
 
+local function check_self_type(self, func)
+  assert(type(self) == "table", ("Use date:%s(...) instead of date.%s(...)"):format(func, func))
+end
+
 function _M:get(field)
-  assert(field)
+  assert(field, "Field is not specified")
+  check_self_type(self, "get")
+  assert(self.cal, "Calendar is not specified")
```

Comment:
This could be part of `check_self_type`.

Reply:
Done